### PR TITLE
Feature/2002 reshape test

### PR DESCRIPTION
**Pull request by chris-chris**
Add 4 pytests for module, moving_averages, and reshape.

### CODE REVIEW

File: haiku/_src/moving_averages_test.py [hunk 1]
```diff
@@ -53,6 +53,17 @@ def test_warmup(self):
     np.testing.assert_allclose(
         (random_input[0] + random_input[1]) / 2, ema(random_input[1]))
 
+  def test_invalid_warmup_length(self):
```
Comment (tomhennigan) [inline haiku/_src/moving_averages_test.py]: I think this needs `@test_utils.transform_and_run` too.

File: haiku/_src/reshape_test.py [hunk 2]
```diff
@@ -83,5 +83,10 @@ def test_flatten_nd(self):
     y = mod(x)
     self.assertEqual(x.shape, y.shape)
 
+  def test_flatten_invalid_preserve_dims(self):
```
Comment (tomhennigan) [inline haiku/_src/reshape_test.py]: Same here, `@test_utils.transform_and_run` please.

File: haiku/_src/moving_averages_test.py (hunk 1 above)
Comment (chris-chris) [reply haiku/_src/moving_averages_test.py]: ok :)

File: haiku/_src/reshape_test.py (hunk 2 above)
Comment (chris-chris) [reply haiku/_src/reshape_test.py]: +1

**Comment by tomhennigan** [comment]
Sorry for the delay here, I didn't get a chance to setup CI today, I will try and find time this weekend or first thing next week!

**Comment by trevorcai** [comment]
Hey, some stuff is triggering Lint errors:
- module_test.py: use " instead of ' for strings
- moving_averages_test.py: indent continuation lines by 4 spaces, not 2

@tomhennigan: we should expose a pylintrc for external contributors to use so we don't need to check this internally.

**Comment by trevorcai** [comment]
Otherwise, changes look great, thanks for contributing :)

**Comment by chris-chris** [comment]
@trevorcai Thank you!
Would you mind if tell me your lint style?
Is it the same with the one of Tensorflow?
https://www.tensorflow.org/community/contribute/code_style

**Comment by inoryy** [comment]
Tiny nit-pick: try to keep commit messages descriptive of the changes and in case of feedback changes consider amending & force-pushing instead. It's a non-issue since the maintainer can just squash individual commits before merging, but maybe one less thing to worry about.

**Pull request merged**